src/m4st/translate/model.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+
+from transformers import T5ForConditionalGeneration, T5Tokenizer
+
+from m4st.ollama.client import generate
+
+language_iso_to_name = {
+    "eng": "English",
+    "deu": "German",
+    "fra": "French",
+    "jpn": "Japanese",
+    "spa": "Spanish",
+    "rou": "Romanian",
+    "zho": "Chinese",
+}
+
+
+class TranslationModel(ABC):
+    @abstractmethod
+    def __call__(self, text: str):
+        pass
+
+
+class T5TranslateModel(TranslationModel):
+    def __init__(
+        self,
+        source_lang_iso: str,
+        target_lang_iso: str,
+        model_tag: str = "google-t5/t5-small",
+    ):
+        self.model = T5ForConditionalGeneration.from_pretrained(model_tag)
+        self.tokenizer = T5Tokenizer.from_pretrained(model_tag)
+
+        self.supported_languages: list[str] = ["eng", "deu", "fra"]
```

Comment:
I guess this depends on the provided `model_tag`? Maybe should also be an arg.

Reply:
Similar to mentioned above issue - good spot. The model tag should not be modifiable, and yes, it depends on the specific models, so I'd rather move the tag inside the class, and not expose it as a parameter.